reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -8459,6 +8459,26 @@ public final void subscribe(Subscriber<? super T> actual) {
 		return FluxName.createOrAppend(this, key, value);
 	}
 
+	/**
+	 * Tag this flux with a key/value pairs. These can be retrieved as a {@link Set} of
+	 * all tags throughout the publisher chain by using {@link Scannable#tags()} (as
+	 * traversed
+	 * by {@link Scannable#parents()}).
+	 * <p>
+	 * Note that some monitoring systems like Prometheus require to have the exact same set of
+	 * tags for each meter bearing the same name.
+	 *
+	 * @param tags are key value pairs
+	 *
+	 * @return the same sequence, but bearing tags
+	 *
+	 *@see #name(String)
+	 *@see #metrics()
+	 */
+	public final Flux<T> tag(Set<Tuple2<String, String>> tags) {
```

Comment:
as discussed above, use `Map<String, String>`. a similar method will need to be added to `ParallelFluxName`